### DISCUSSION TITLE
Metamask v2 Status

### DISCUSSION THREAD

**Discussion by pqseags** [WalletConnect/walletconnect-monorepo#2662] [Developer Support] [answered]
As the v2 migration deadline gets extremely close, I'm sure many are wondering about the status of Metamask v2 support. From what I remember there was a publicly communicated date of June 15 which has now passed and Metamask is still not supported. 

As a dapp that uses WalletConnect and wants to support v2, we obviously cannot migrate without Metamask support. I also cannot find any clear communication regarding the status. I'm starting this thread in hopes of consolidating information and updates. Communication from the WalletConnect or Metamask teams on this would be greatly appreciated.

**Comment by thmsl** [comment]
According to [this page](https://docs.walletconnect.com/2.0/advanced/migration-from-v1.x/migration-faq#when-will-walletconnect-v10-be-shut-down---and-what-does-that-mean), v1 will stop working on June 28

> We will shut down WalletConnect v1.0 at 2pm (UTC) on June 28, 2023. Wallets and apps still utilizing the WalletConnect v1.0 bridge after this date will no longer be able to connect their users to another WalletConnect-supported app.

V2 seems to have issues on iOS with Trust wallet (at least) and still no metamask support.
We are then considering keeping both v1 and v2, but useless if v1 is shut down.

Would be great to know officially if the deadline can be pushed back

**Reply by glitch-txs** [reply]
Hi, what issues does Trust Wallet have on iOS?

**Reply by thmsl** [reply]
Using iOS Safari, Trust v8.8
We tried several implementations of V2  (https://etherscan.io/verifiedSignatures, https://react-app.walletconnect.com/)
Wallet connection is working but signature returns an error `unable to decrypt exception`

See screenshot from etherscan
![image (21)](https://github.com/WalletConnect/walletconnect-monorepo/assets/496215/211eba79-bc9c-494d-8cff-0add37780938)

Shall I create a new thread?

**Reply by glitch-txs** [reply]
It would be nice, could you add a minimal reproducible example as well?

**Comment by m1heng** [comment]
TBH, if v1 is shut down without MetaMask supporting v2, this would totally be a nightmare for WEB3 world

**Comment by siosio34** [comment]
I am using WAGMI and walletconnectConnector now.

 From June 15, metamask used to migrate to be able to use Wallet Connect 2, but Work was slow on the metamask side and I finally tried it today.
 
In the case of launching Metamask QR in Wallet Connect on the desktop, the connection works well, but 
It seems that there are many bugs in the mobile web browser, such as the signature not appearing often and the original page not returning well.


As such, if WalletConnect1 ends support on the 28th, a lot of confusion will arise.


As far as I know, the most convenient way to connect a metamask wallet in a mobile web browser is to use WalletConnect, but that method is currently not working well. and Existing users who have not updated metamask know that even if they provide a connection with walletconnectv2, they cannot actually connect.

Until the new version is stabilized, I think a little more process of operating in parallel with the existing version is needed.

Thank you for listening. have a good day

**Comment by cryptopro111** [comment]
We have implemented the V1 on our product and about the launch, just really worry about if the V1 will be closed directly after V2 launch, it seems that V2 is not so ready to be used. I hope that wallet connect can release some update news about the arrangement. I am sure many developers and developments are waiting for further action

**Comment by pqseags** [comment]
Following up - why is this question not being answered, while even subquestions in the same thread are? The number of upvotes and comments on this post make it clear that people are seeking answers.

Even if the answer is "Metamask is not going to be supported in v2 at the time of v1 shutdown", that would still be helpful so that we know we need to choose a new plan of action. Without any official update or guidance we don't know how to proceed. Again, ANY communication would be extremely helpful!

**Reply by glitch-txs** [reply]
Hi, MetaMask already supports v2 on its latest version

**Reply by pqseags** [reply]
Thanks for the response! Is this true on iOS? We're building an iOS application that uses the WalletConnect Swift SDK to connect with the Metamask iOS app.

**Reply by glitch-txs** [reply]
Unfortunately I was only able to test on Android, maybe someone else can confirm this. I think they use RN though right?

**Reply by pqseags** [reply]
I'm not sure what they use under the hood, but their iOS app hasn't been updated in the App Store for 2 months. It could be migrated in the backend potentially but it seems unlikely? During our migration attempt over the last few weeks it seemed that Metamask was not supported/working.

**Reply by arein** [reply, marked as answer]
MetaMask iOS is expected to be released shortly

**Reply by pqseags** [reply]
That's great to hear! Is there any more clarity around when 'shortly' might be?

**Reply by heebaebang** [reply]
When I try to connect my wallet with the Android Metamask app [here](https://react-auth-dapp.walletconnect.com/), there is no response. Metamask version is 7.0.1. What could be the problem?

**Reply by finessevanes** [reply]
@pqseags MetaMask is now live on iOS as of Friday, June 23. You will need to manually go into the app store, tap on MetaMask, then tap on update.

**Reply by pqseags** [reply]
Thank you for the heads up! Really glad it made it in time!

**Reply by hammadtariq98** [reply]
@finessevanes After updating metaMask still does not connect wallet on both devices IOS and Android.

example app [link](https://mv7snq.csb.app/)

**Comment by finessevanes** [comment]
> When I try to connect my wallet with the Android Metamask app [here](https://react-auth-dapp.walletconnect.com/), there is no response. Metamask version is 7.0.1. What could be the problem?

Please open a separate issue or discussion for this 🙏🏽

**Comment by ahollenbach** [comment]
Now that Android + iOS Metamask 7.0.1 is out, requesting Metamask support be officially marked for v2 ([it's still v1 only in the API](https://walletconnect.com/explorer/metamask)). Without this, when a user scans the v2 QR code with the latest version of the app, the session doesn't seem to get properly stored and subsequent requests (ie for a sign) don't work:

```
Failed to launch 'zerion://wc?requestId=<...>' because the scheme does not have a registered handler.
```

(Zerion was the last successful wallet connection)

Removing `WALLETCONNECT_DEEPLINK_CHOICE` and `W3M_RECENT_WALLET_DATA` resolved the issue, but I'd rather not mess with that if avoidable.
